pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -200,3 +200,166 @@ func newFastEcAddAssignNewYHint() hinter.Hinter {
 func createFastEcAddAssignNewYHinter() (hinter.Hinter, error) {
 	return newFastEcAddAssignNewYHint(), nil
 }
+
+func newFastEcAddAssignNewXHint(slope, point0, point1 hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "FastEcAddAssignNewX",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//
+			//> slope = pack(ids.slope, PRIME)
+			//> x0 = pack(ids.point0.x, PRIME)
+			//> x1 = pack(ids.point1.x, PRIME)
+			//> y0 = pack(ids.point0.y, PRIME)
+			//> value = new_x = (pow(slope, 2, SECP_P) - x0 - x1) % SECP_P
+
+			slopeAddr, err := slope.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			slopeMemoryValues, err := hinter.GetConsecutiveValues(vm, slopeAddr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			point0Addr, err := point0.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			point0MemoryValues, err := hinter.GetConsecutiveValues(vm, point0Addr, int16(6))
+			if err != nil {
+				return err
+			}
+
+			point1Addr, err := point1.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			point1MemoryValues, err := hinter.GetConsecutiveValues(vm, point1Addr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			// [d0, d1, d2]
+			var slopeValues [3]*fp.Element
+			// [x.d0, x.d1, x.d2]
+			var point0XValues [3]*fp.Element
+			// [y.d0, y.d1, y.d2]
+			var point0YValues [3]*fp.Element
+			// [x.d0, x.d1, x.d2]
+			var point1XValues [3]*fp.Element
+
+			for i := 0; i < 6; i++ {
+				if i < 3 {
+					slopeValue, err := slopeMemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					slopeValues[i] = slopeValue
+
+					point0XValue, err := point0MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point0XValues[i] = point0XValue
+
+					point1XValue, err := point1MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point1XValues[i] = point1XValue
+				} else {
+					point0YValue, err := point0MemoryValues[i].FieldElement()
+					if err != nil {
+						return err
+					}
+					point0YValues[i-3] = point0YValue
+				}
```

Comment:
For cases like this, it looks cleaner if you just do two separate for loops, one from i = 0 to 3 and the other from i = 3 to 6. You'll also do less operations, currently you always have to perform the if question.